connectors/agent/protocol.py

```diff
@@ -0,0 +1,68 @@
+from elastic_agent_client.generated import elastic_agent_client_pb2 as proto
+from elastic_agent_client.handler.action import BaseActionHandler
+from elastic_agent_client.handler.checkin import BaseCheckinHandler
+from elastic_agent_client.util.logger import logger
+
+
+class ConnectorActionHandler(BaseActionHandler):
+    """Class handling Agent actions.
+
+    As there are no actions that we can respond to, we don't actually do anything here.
+    """
+
+    async def handle_action(self, action: proto.ActionRequest):
+        """Implementation of BaseActionHandler.handle_action
+
+        Right now does nothing as Connectors Service has no actions to respond to.
+        """
+        msg = (
+            f"This connector component can't handle action requests. Received: {action}"
+        )
```

Comment:
should we include the error log? I don't see any other "wrapper" logic that would surface this to the user / logs

Reply:
It should log the error. See: https://github.com/elastic/python-elastic-agent-client/blob/ba2525d23ed424c5740d9bbf87baff80a90533e7/elastic_agent_client/service/actions.py#L51-L61

It won't terminate, no. It just sends a message back through the protocol that the requested action failed. See this slack thread: https://elastic.slack.com/archives/C01QQ449KE1/p1719235054818189?thread_ts=1718987763.249969&cid=C01QQ449KE1